src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -146,7 +146,13 @@ public function getTypes($class, $property, array $context = []): ?array
         /** @var DocBlock\Tags\Var_|DocBlock\Tags\Return_|DocBlock\Tags\Param $tag */
         foreach ($docBlock->getTagsByName($tag) as $tag) {
             if ($tag && !$tag instanceof InvalidTag && null !== $tag->getType()) {
-                $types = array_merge($types, $this->phpDocTypeHelper->getTypes($tag->getType()));
+                foreach ($this->phpDocTypeHelper->getTypes($tag->getType()) as $type) {
+                    if (Type::BUILTIN_TYPE_OBJECT === $type->getBuiltinType() && 'self' === $type->getClassName()) {
```

Comment:
what about "static" and "parent"?

Reply:
`static` is something that would need to be evaluated at runtime. Not sure how to deal with that here.

Reply:
true, let's leave it as is - only "parent" and "self" can be resolved